tools/webpack

```diff
@@ -55,7 +57,22 @@ def build_for_dev_server(host, port, minify, disable_host_check):
         webpack_args.append('--optimize-minimize')
     if disable_host_check:
         webpack_args.append('--disable-host-check')
-    os.execvp(webpack_args[0], webpack_args)
+    webpack_process = subprocess.Popen(webpack_args)
+
+    class WebpackConfigFileChangeHandler(pyinotify.ProcessEvent):
+        def process_default(self, event):
+            # type: (Any) -> None
+            nonlocal webpack_process
+            print('Restarting webpack-dev-sever due to config changes...')
+            webpack_process.kill()
+            webpack_process = subprocess.Popen(webpack_args)
+
+    watch_manager = pyinotify.WatchManager()
+    event_notifier = pyinotify.Notifier(watch_manager, WebpackConfigFileChangeHandler())
+    for file in ['webpack.config.ts', 'webpack-helpers.ts', 'webpack.assets.json']:
+        filepath = os.path.join(os.path.dirname(__file__), file)
+        watch_manager.add_watch(filepath, pyinotify.IN_MODIFY)
+    event_notifier.loop()
```

Comment:
This should all go in a `try … finally` block that ensures we `webpack_process.terminate(); webpack_process.wait()` at the end.

Reply:
Just to make sure, this is for making sure webpack process exits cleanly at the end? )(Also curious to why we will need to do this when we weren't doing this before)

Reply:
Right. It wasn’t needed before because `os.execvp` replaces the current process with the `webpack` process and does not return to the current script.